modules/test/playwright/tests/frontend-data-set-web/main/advanced.spec.ts

```diff
@@ -1161,6 +1161,169 @@ test(
 	}
 );
 
+test(
+	'Test quick action buttons function correctly',
+	{tag: '@LPS-153220'},
+	async ({fdsSamplePage, page}) => {
+		const firstRowItemActionButton = fdsSamplePage.table.itemActionsCells
+			.first()
+			.getByRole('button', {
+				exact: true,
+				name: 'Actions',
+			});
+
+		const thirdRowItemActionButton = fdsSamplePage.table.itemActionsCells
+			.nth(2)
+			.getByRole('button', {
+				exact: true,
+				name: 'Actions',
+			});
+
+		const firstRowSampleEditQuickActionLink = fdsSamplePage.table.bodyRows
+			.first()
+			.getByRole('link', {
+				name: 'Sample Edit',
+			});
+
+		const thirdRowSampleEditQuickActionLink = fdsSamplePage.table.bodyRows
+			.nth(2)
+			.getByRole('link', {
+				name: 'Sample Edit',
+			});
+
+		const firstTableHeadCell = fdsSamplePage.table.headerCells.first();
+
+		await test.step('Assert that clicking quick action is equivalent to clicking the ellipsis dropdown menu', async () => {
+			await firstRowItemActionButton.hover();
+
+			await fdsSamplePage.table.bodyRows
+				.first()
+				.getByRole('link', {
+					name: 'Sample Edit',
+				})
+				.click();
+
+			const pageURLAfterQuickAction = page.url();
+
+			expect(pageURLAfterQuickAction).toContain('#test-pencil');
+
+			await firstRowItemActionButton.click();
+
+			await page
+				.getByRole('menuitem', {
+					name: 'Sample Edit',
+				})
+				.click();
+
+			expect(page.url()).toEqual(pageURLAfterQuickAction);
+		});
+
+		await test.step('Assert that hover over mouse off of the table body quick action menu is not visible', async () => {
+			await firstRowItemActionButton.hover();
+
+			await expect(firstRowSampleEditQuickActionLink).toBeVisible();
+
+			await firstTableHeadCell.hover();
+
+			await expect(firstRowSampleEditQuickActionLink).not.toBeVisible();
+		});
+
+		await test.step('When hovering over the first line item and the quick action menu is displayed on the 1st line', async () => {
+			const firstTableRow = fdsSamplePage.table.bodyRows.first();
+
+			await firstTableRow.hover();
+
+			await expect(firstRowSampleEditQuickActionLink).toBeVisible();
+		});
+
+		await test.step('When clicking on the ellipsis and hovering over another row, multiple quick action menus are displayed', async () => {
+			await thirdRowItemActionButton.hover();
+
+			await expect(thirdRowSampleEditQuickActionLink).toBeVisible();
+
+			await thirdRowItemActionButton.hover();
+
+			await thirdRowItemActionButton.click();
+
+			await expect(page.locator('.dropdown-menu.show')).toBeVisible();
+
+			await fdsSamplePage.table.bodyRows.first().hover();
+
+			await expect(
+				page
+					.locator('.fds table tbody tr')
+					.first()
+					.getByLabel('Sample Edit')
+			).toBeVisible(); // `page` must be used here since using `fdsSamplePage` won't locate the element with the dropdown open
```

Comment:
With the dropdown open, I'm not quite sure why `fdsSamplePage.table.bodyRows` won't locate the element but writing it out with `page` will so I added this note here.

Reply:
I am not sure exactly how PW internally works, but my mental model is that that specific pages like `fdsSamplePage` behave like instances. When you make a change like `fdsSamplePage.someElement.someAction()`, it changes instance state. When you make a change like `page.someElement.someAction()`, `fdsSamplePage` instance does not change. When you reference it next time, it abruptly has some old state. My main suspect here would be this from few steps above.
```js
	await fdsSamplePage.table.bodyRows
		.first()
		.getByRole('link', {
			name: 'Sample Edit',
		})
		.click();
...

	await page
		.getByRole('menuitem', {
			name: 'Sample Edit',
		})
		.click();
```
Of course, I might be totally off mark here :sweat_smile:

Reply:
So after playing around a bit more I found that it was `getByRole` that would have issues locating when the dropdown was open. Originally the variable was defined as:

```
const firstRowSampleEditQuickActionLink = fdsSamplePage.table.bodyRows
	.first()
	.getByRole('link', {
		name: 'Sample Edit',
	});
```

Switching this to `getByLabel` would be able to consistently locate the element. 🙃 

```
const firstRowSampleEditQuickActionLink = fdsSamplePage.table.bodyRows
	.first()
	.getByLabel('Sample Edit');
```

Anyway, I was able to remove this `page` use/comment here.